code-challenges.test.js

```diff
@@ -28,30 +37,119 @@ const colors2 = [
 
 // Pseudo code:
+//Function Name: totalVotes
+//Input: Objects
+//Output: Number
+//Process: Create a function called totalVotes. Use the argument object.valueName to get just the number from the object. Then subtract it to get the final answer. 
+
+const totalVotes = (object) => {
+  return object.upVotes - object.downVotes
+}
+
+// console.log(totalVotes(votes1))
+// console.log(totalVotes(votes2))
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
+describe("fixArray", () => {
+  it("Returns an array that should repeat words from previous two arrays, but only have the same words once on the new array.", () => {
+    expect(fixArray(...dataTypesArray1,...dataTypesArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+  })
+})
 
 const dataTypesArray1 = ["array", "object", "number", "string", "Boolean"]
 const dataTypesArray2 = ["string", "null", "Boolean", "string", "undefined"]
+
 // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+//Function Name: fixArray
+//Input: Two arrays
+//Output: One array
+//Process: Create a function called fixArray. Take in two arrays in the argument. Create a variable and within it use filter to filter through an a array and have it check what values we don't want duplicated. Return the value of everything that does not include "string" and "Boolean". Outside of the .filter return both arrays combined. 
+
+//First Try
+// Could not get it to run with JEST! Could you explain, please. 
+// const fixArray = (arr1, arr2) => {
+//    const filterArr = arr2.filter((value) => {
+//     return ! value.includes("string") && ! value.includes("Boolean")
+//   })
+//   return arr1.concat(filterArr)
+// }
+//console.log(fixArray(...dataTypesArray1,...dataTypesArray2))
+
+// Pseudo code:
+//Function Name: fixArray
+//Input: Two arrays
+//Output: One array
+//Process: Create a function called fixArray. Use ...to take in iteration of an array or string. Filter through the array and check the values as well as the index. When iterating through the array we will return an array which will return the values of the array, but if there is a duplication then the value will not be returned. 
+
+//Second Try
+//Had to get help, but would my first try work?
+const fixArray = (...arr) => {
+  return arr.filter((value, index) => {
+    return arr.indexOf(value) == index //If the value appears more than once then .indexOf will take the first word of it. 
+  })
+}
+// console.log(fixArray(...dataTypesArray1, ...dataTypesArray2))
+
+//Don't really understand the spread operator. Could you explain it a little more?
```

Comment:
The spread operator is a tool used to "spread" the elements of any iterable (such as arrays, strings, objects). So in this challenge, we are using it on an array.
Here's a basic example of how it works. Let's say I have this array: `const originalArray = [1, 2, 3]`
I can spread the contents of that array into another array like this: `const copyArray = [...originalArray]`.
When we do this, the `copyArray` variable is now the exact same array as `originalArray`. We are taking the elements in `originalArray` and putting them into a separate array `copyArray`.

We can also combine arrays using this method like so:
`const combinedArray = [...originalArray, ...copyArray]`
Both `originalArray` and `copyArray` are the same array: `[1, 2, 3]`
For `combinedArray`, we are taking the elements of both of those arrays, and putting them into a new array.
If we were to console.log(combinedArray), we would get: `[1, 2, 3, 1, 2, 3]`

You can also think of it as taking the contents that are already in whatever you are using it on, and copying them into something else.

For this challenge, when you put `...arr` in the parameter field, it will take the arrays that the function gets as arguments and put the arrays into another array, so if you passed in 2 arrays as arguments, you would then have an array with 2 arrays inside of it. The stretch challenge was figuring out how to get rid of nested arrays to work with a single array again.